docroot/profiles/custom/cgov_site/modules/custom/cgov_blog/cgov_blog.module

```diff
@@ -218,3 +219,32 @@ function cgov_blog_preprocess_block(&$variables) {
   }
 
 }
+
+/**
+ * Implements hook_preprocess_node().
+ */
+function cgov_blog_preprocess_node(&$variables) {
```

Comment:
@kalidindir We try not to use preprocess hooks as there are usually better ways. The blog series cache tag should really be set the the various block plugins in the cgov_blog module. Those blocks are what actually depend on the series. If I were to remove those blocks from the blog_post displays, then the blog_post no longer has a cache dependency on the blog series. Additionally, by adding it on the blocks, you do not have to specifically check for the view mode of the post. This is even better if we ever create a separate view_mode that contains one of those blocks we would not have to update this logic.

The taxonomy_term_update looks pretty good and has nice comments. There is just one change I would make and I will add it there.